data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -258,7 +260,21 @@ public void doOutput(final Collection<Record<Event>> records) {
   private SerializedJson getDocument(final Event event) {
     String docId = (documentIdField != null) ? event.get(documentIdField, String.class) : null;
     String routing = (routingField != null) ? event.get(routingField, String.class) : null;
-    return SerializedJson.fromStringAndOptionals(event.toJsonString(), docId, routing);
+
+    final String document = buildRawDocument(event);
+
+    return SerializedJson.fromStringAndOptionals(document, docId, routing);
+  }
+
+  private String buildRawDocument(final Event event) {
+    if (documentRootKey != null && event.containsKey(documentRootKey)) {
+      final String document = event.getToJsonString(documentRootKey);
+      if (document == null || !document.startsWith("{")) {
```

Comment:
I'm wondering about this behavior. It could be confusing.

I have an event with:

```
{
  "myRoot" : {
     "a": "b",
     "c" : "d"
   },
   "willIgnore" : "does not matter"
}
```

My document looks like:

```
{ 
  "a" : "b",
  "c" : "d"
}
```

But, if if I have:

```
{
  "myRoot" : "hello world",
  "willIgnore" : "does not matter"
}
```

I get 

```
{
  "myRoot" : "hello world"
}
```

This seems quite a bit different since it now includes the name of the root key.

Also, what if the `document_root_key` is a JSON path?

Reply:
I originally had `data` as the default key if the value is a basic type but thought the provided `document_root_key` made the most sense. If the key a JSON path, the full path would be here. That's probably not what we want either. I will revert back to `data` as the key.